src/helpers.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Container\Container;
+use MikeFrancis\LaravelSecureHeaders\Exceptions\ContentSecurityPolicyNotFoundException;
+
+if (!function_exists('csp_nonce')) {
+    /**
+     * The nonce will be regenerated once per page load.
+     *
+     * @param string $friendlyDirective
+     * @return string
+     * @throws ContentSecurityPolicyNotFoundException
+     */
+    function csp_nonce(string $friendlyDirective)
+    {
+        $app = Container::getInstance();
+        $request = $app['request'];
+        $cspHeader = $request->headers->get('content-security-policy');
+
+        if (!$cspHeader) {
+            throw new ContentSecurityPolicyNotFoundException();
+        }
+
+        $contentTypes = explode('; ', $cspHeader);
+
+        foreach ($contentTypes as $contentType) {
+            preg_match("/([a-z]+)-src 'nonce-([^']+)'/", $contentType, $matches);
```

Comment:
You'd want to deal with the case that the nonce isn't the first source item, also the user may use `script-src` with the function instead of just `script` for example. It might be easier to draw a line and only support the full names of directives here – as in future it might be the case that directives that do not end in `-src` have associated nonces. (but at least things like `script-src` if used directly should work for now).
Of course, this would also break in the case that `script-src` was included as a source list item within some other directive that occurred before for the actual `script-src` for example.

I'd suggest using something like the following on the entire string:

```regex
(?:;|^)[ ]*+script-src[^';]*+(?<=[ ])'nonce-([^'; ]++)'
```
[quick test](https://regex101.com/r/oOi3jO/1/)

But replace `script-src` with the `preg_quote` of the directive supplied by the user.
Just as a general point, it can help performance to add `+` after your regex quantifiers to make them possessive if they don't need backtracking :) No backtracking will cause them to fail quickly instead of running backwards trying to find a match if you know they won't work other than on the greediest match.
(You can do this when the set of things with a quantifier cannot intersect with what follows it – so for example things like `[^']*'` or `[ ]*s` can be made possessive because no not-quote can be a quote, and a space can't be an `s`).

Reply:
If you need a test case, try pulling the nonce out of this :)
https://securityheaders.io/?q=aidanwoods.com&followRedirects=on

(you can probably find some more by browsing recent scans)